interpreter/value_storage_reference.go

```diff
@@ -230,21 +230,11 @@ func (v *StorageReferenceValue) GetMember(context MemberAccessibleContext, locat
 	// It is not possible (or a lot of work), to create the bound function with the storage reference
 	// when it was created originally, because `getMember(referencedValue, ...)` doesn't know
 	// whether the member was accessed directly, or via a reference.
-
-	if boundFunction, isBoundFunction := member.(BoundFunctionValue); isBoundFunction {
-		referencedValueStaticType := referencedValue.StaticType(context)
-
-		boundFunction.SelfReference = NewStorageReferenceValue(
-			context,
-			v.Authorization,
-			v.TargetStorageAddress,
-			v.TargetPath,
-			context.SemaTypeFromStaticType(referencedValueStaticType),
-		)
-		return boundFunction
-	}
-
-	return member
+	return context.MaybeUpdateStorageReferenceMemberReceiver(
+		v,
+		referencedValue,
+		member,
+	)
```

Comment:
The member's receiver was updated if the member is a bound function. However, interpreter and VM has two implementations for bound functions. Hence, delegate this receiver-update logic to the corresponding implementation (i.e: to the context)